### PR TITLE
py-deeptools: add 3.5.2, update dependency versions

### PR DESCRIPTION
**Pull request by A-N-Other**
Updating to `@3.5.2`. Updating the dependency requirements as per its `setup.py`. No other changes required to get this installed for me.

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-deeptools/package.py [hunk 1]
```diff
@@ -25,9 +26,11 @@ class PyDeeptools(PythonPackage):
     depends_on("py-py2bit@0.2.0:", type=("build", "run"))
     depends_on("py-pybigwig@0.2.1:", type=("build", "run"))
     depends_on("py-pysam@0.14.0:", type=("build", "run"))
-    depends_on("py-matplotlib@2.1.2:", type=("build", "run"))
+    depends_on("py-matplotlib@3.3.0:", type=("build", "run"), when="@3.5:")
```
Comment (manuelakuhn) [inline var/spack/repos/builtin/packages/py-deeptools/package.py]:
```suggestion
    depends_on("py-matplotlib@3.3.0:", type=("build", "run"), when="@3.5.1:")
```

File: var/spack/repos/builtin/packages/py-deeptools/package.py [hunk 2]
```diff
@@ -25,9 +26,11 @@ class PyDeeptools(PythonPackage):
...
+    depends_on("py-matplotlib@2.1.2:", type=("build", "run"), when="@:3.3.0")
     depends_on("py-numpydoc@0.5:", type=("build", "run"))
-    depends_on("py-plotly@2.0.0:", type=("build", "run"))
+    depends_on("py-plotly@4.9:", type=("build", "run"), when="@3.5.1")
```
Comment (manuelakuhn) [inline var/spack/repos/builtin/packages/py-deeptools/package.py]:
```suggestion
    depends_on("py-plotly@4.9:", type=("build", "run"), when="@3.5.1:")
```

**Review by manuelakuhn — COMMENTED** [review]
If you want to make this package easier to review in the future you can 
- remove old unsupported python version, here the python@2.7 dependency (spack does not support python@3.6 and older)
- sort the dependencies in the same order as they are listed in the package (here setup.py)
- put `when` before `type` to make it more consistent with other python packages

**Review by adamjstewart — DISMISSED** [review]
I'm fine with this as is, but I agree that @manuelakuhn's suggestions are more technically correct. Let me know whether you want to update or just merge.

**Comment by A-N-Other** [comment]
Changes made as suggested, @manuelakuhn @adamjstewart

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**